src/nestegg.c

```diff
@@ -834,7 +834,15 @@ ne_read_float(nestegg_io * io, double * val, uint64_t length)
 {
   union {
     uint64_t u;
-    float f;
+    struct {
+#if __FLOAT_WORD_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
Thanks.  Yeah, it's possible this won't be defined on some particular platform/compiler but the most common/likely ones are covered.  In the event that this fix breaks for a big-endian system with a compiler that doesn't define __FLOAT_WORD_ORDER__, it is covered by the included regression tests (e.g. detodos.webm will fail) so it can, at least, be found and fixed quickly.